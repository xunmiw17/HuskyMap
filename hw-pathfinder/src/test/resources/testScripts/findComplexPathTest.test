CreateGraph simpleGraph

AddNode simpleGraph A
AddNode simpleGraph B
AddNode simpleGraph C
AddNode simpleGraph D
AddNode simpleGraph E
AddNode simpleGraph F
AddNode simpleGraph G
AddNode simpleGraph H

AddEdge simpleGraph A B 2.0
AddEdge simpleGraph A C 1.0
AddEdge simpleGraph A D 4.0
AddEdge simpleGraph B C 5.0
AddEdge simpleGraph B E 10.0
AddEdge simpleGraph B F 2.0
AddEdge simpleGraph C A 9.0
AddEdge simpleGraph C E 11.0
AddEdge simpleGraph D C 2.0
AddEdge simpleGraph E D 7.0
AddEdge simpleGraph E G 1.0
AddEdge simpleGraph F H 3.0
AddEdge simpleGraph G E 3.0
AddEdge simpleGraph G F 2.0
AddEdge simpleGraph H G 1.0

FindPath simpleGraph A B

FindPath simpleGraph A C

FindPath simpleGraph A D

FindPath simpleGraph A E

FindPath simpleGraph A F

FindPath simpleGraph A G

FindPath simpleGraph A H