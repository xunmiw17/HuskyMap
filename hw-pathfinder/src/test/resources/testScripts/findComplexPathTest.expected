created graph simpleGraph

added node A to simpleGraph
added node B to simpleGraph
added node C to simpleGraph
added node D to simpleGraph
added node E to simpleGraph
added node F to simpleGraph
added node G to simpleGraph
added node H to simpleGraph

added edge 2.000 from A to B in simpleGraph
added edge 1.000 from A to C in simpleGraph
added edge 4.000 from A to D in simpleGraph
added edge 5.000 from B to C in simpleGraph
added edge 10.000 from B to E in simpleGraph
added edge 2.000 from B to F in simpleGraph
added edge 9.000 from C to A in simpleGraph
added edge 11.000 from C to E in simpleGraph
added edge 2.000 from D to C in simpleGraph
added edge 7.000 from E to D in simpleGraph
added edge 1.000 from E to G in simpleGraph
added edge 3.000 from F to H in simpleGraph
added edge 3.000 from G to E in simpleGraph
added edge 2.000 from G to F in simpleGraph
added edge 1.000 from H to G in simpleGraph

path from A to B:
A to B with weight 2.000
total cost: 2.000

path from A to C:
A to C with weight 1.000
total cost: 1.000

path from A to D:
A to D with weight 4.000
total cost: 4.000

path from A to E:
A to B with weight 2.000
B to F with weight 2.000
F to H with weight 3.000
H to G with weight 1.000
G to E with weight 3.000
total cost: 11.000

path from A to F:
A to B with weight 2.000
B to F with weight 2.000
total cost: 4.000

path from A to G:
A to B with weight 2.000
B to F with weight 2.000
F to H with weight 3.000
H to G with weight 1.000
total cost: 8.000

path from A to H:
A to B with weight 2.000
B to F with weight 2.000
F to H with weight 3.000
total cost: 7.000